src/App.tsx

```diff
@@ -26,70 +27,128 @@ type ReorderOptions = {
+    case SortType.ALPHABET:
+      visibleGoods.sort();
+      break;
+    case SortType.LENGTH:
+      visibleGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+    default:
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    sortType: SortType.NONE,
+    isReversed: false,
+  };
+
+  sortByAlphabetHandler = () => {
+    this.setState({ sortType: SortType.ALPHABET });
+  };
+
+  sortByLengthHandler = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  reverseHandler = () => {
+    this.setState(prevState => ({
+      isReversed: !prevState.isReversed,
+    }));
+  };
+
+  resetHandler = () => {
+    this.setState({
+      isReversed: false,
+      sortType: SortType.NONE,
+    });
+  };
+
+  render() {
+    const visibleGoods = getReorderedGoods(goodsFromServer, {
+      sortType: this.state.sortType,
+      isReversed: this.state.isReversed,
+    });
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames({
+              button: true,
+              'is-info': true,
+              'is-light': this.state.sortType !== SortType.ALPHABET,
+            })}
+            onClick={() => this.sortByAlphabetHandler()}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={classNames({
+              button: true,
+              'is-success': true,
+              'is-light': this.state.sortType !== SortType.LENGTH,
+
+            })}
+            onClick={() => this.sortByLengthHandler()}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={classNames({
+              button: true,
+              'is-warning': true,
+              'is-light': !this.state.isReversed,
+
+            })}
+            onClick={() => this.reverseHandler()}
```

Comment:
```suggestion
            onClick={this.reverseHandler}
```

fix it in the rest of the code
if your method does not required argument, you don't have to write it as arrow function that calls your method, simply pass your method instead